### PR TITLE
JSON v2 for RxPk

### PR DESCRIPTION
**Pull request by lthiery**
[json_diffs.txt](https://github.com/helium/semtech-udp/files/5455019/json_diffs.txt)

Lazily implemented the extra fields documented here just so RXPK v2 would work. Therefore, some of the extra data is thrown away from the JSON v2

### CODE REVIEW

File: src/packet/parser.rs [hunk 1]
```diff
@@ -33,7 +33,8 @@ impl Parser for Packet {
                 Identifier::PushData => {
                     let gateway_mac = gateway_mac(&buffer[4..12]);
                     let json_str = std::str::from_utf8(&buffer[12..num_recv])?;
-                    let data = serde_json::from_str(json_str)?;
+                    let data = serde_json::from_str(json_str).unwrap();
```
Comment (madninja) [inline src/packet/parser.rs:36]: This would panic with the unwrap?

File: src/packet/push_data.rs [hunk 2]
```diff
@@ -93,21 +93,141 @@ size | number | RF packet payload size in bytes (unsigned integer)
...
+}
+
+#[derive(Debug, Serialize, Deserialize, Clone)]
+#[serde(untagged)]
+pub enum RxPk {
+    V1(RxPkV1),
+    V2(RxPkV2),
+}
+
+macro_rules! get_v1v2 {
```
Comment (madninja) [inline src/packet/push_data.rs:190]: maybe rename to get_field? the v1v2 name is grating and a bit brittle

**Review by madninja — COMMENTED** [review]
Just a few minor comments

File: src/packet/parser.rs (hunk 1 above)
Comment (lthiery) [reply src/packet/parser.rs:36]: oops i was using that to debug. thanks for catching

**Pull request merged**